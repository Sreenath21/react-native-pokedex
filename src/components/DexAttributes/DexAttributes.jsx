import React, {useContext} from 'react';
import {View, Text} from 'react-native';

import {PokemonContext} from '../../context/PokemonContext';
import DexTile from '../DexTile/DexTile';
import {
  getHeight,
  stringifyGender,
  stringifyAbilites,
} from '../../utils/helplerFunctions';
import styles from './DexAttributes.style';

const DexAttributes = () => {
  const {
    pokemon: {height, weight, genders, eggGroups, abilities, types, weaknesses},
  } = useContext(PokemonContext);

  return (
    <View style={styles.container}>
      <View style={styles.twoColumns}>
        <DexTile attribute="Height" value={getHeight(height)} />
        <DexTile attribute="Weight" value={`${weight / 10} Kg`} />
      </View>

      <View style={styles.twoColumns}>
        <DexTile attribute={'Gender(s)'} value={stringifyGender(genders)} />
        <DexTile attribute="Egg Groups" value={eggGroups?.join(', ')} />
      </View>

      <View style={styles.twoColumns}>
        <DexTile attribute="Abilities" value={stringifyAbilites(abilities)} />
        <DexTile attribute="Types" value={types.map(type => type.type.name)} />
      </View>

      <View style={styles.oneColumns}>
        <DexTile attribute="Weak Against" value={weaknesses} />
      </View>
    </View>
  );
};

export default DexAttributes;
