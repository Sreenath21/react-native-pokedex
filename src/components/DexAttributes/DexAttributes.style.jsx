import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    marginTop: 48,
  },
  twoColumns: {
    flexDirection: 'row',
    marginVertical: 16,
  },
  oneColumns: {
    marginVertical: 16,
  },
});
