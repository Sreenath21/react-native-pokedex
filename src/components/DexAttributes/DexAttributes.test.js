import React from 'react';
import DexAttributes from './DexAttributes';
import {render} from '@testing-library/react-native';
import PokemonContextProvider from '../../context/PokemonContext';

it('Renders DexAttributes correctly', () => {
  const screen = render(<DexAttributes />, {wrapper: PokemonContextProvider});

  screen.getByText(/height/i);
  expect(screen).toMatchSnapshot();
});
