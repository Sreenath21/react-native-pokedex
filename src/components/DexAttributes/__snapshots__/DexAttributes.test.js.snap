// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders DexAttributes correctly 1`] = `
<View
  style={
    {
      "marginTop": 48,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "rowGap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Height
      </Text>
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
          }
        }
      >
        0'0"
      </Text>
    </View>
    <View
      style={
        {
          "flex": 1,
          "rowGap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Weight
      </Text>
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
          }
        }
      >
        0 Kg
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "rowGap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Gender(s)
      </Text>
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
          }
        }
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "rowGap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Egg Groups
      </Text>
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "rowGap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Abilities
      </Text>
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
          }
        }
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "rowGap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Types
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "gap": 5,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "marginVertical": 16,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "rowGap": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#391363",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Weak Against
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "gap": 5,
          }
        }
      />
    </View>
  </View>
</View>
`;
